interop/xds/server/server.go

```diff
@@ -78,8 +83,47 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for headerValue := range getRPCBehaviorMetadata(ctx) {
+		if rpcStatus := getStatusForRPCBehaviorMetadata(headerValue); rpcStatus != nil {
+			return response, status.Err(rpcStatus.Code(), rpcStatus.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) map[string]bool {
+	var mdRPCBehavior []string
+	if md, ok := metadata.FromIncomingContext(ctx); !ok {
+		logger.Error("failed to receive metadata")
+		return nil
+	} else {
+		mdRPCBehavior = md.Get(xdsinterop.RPCBehaviorMetadataKey)
+	}
+
+	rpcBehaviorMetadata := make(map[string]bool)
+	for _, val := range mdRPCBehavior {
+		splitVals := strings.Split(val, ",")
+
+		for _, behavior := range splitVals {
+			rpcBehaviorMetadata[strings.TrimSpace(behavior)] = true
+		}
+	}
+	return rpcBehaviorMetadata
+}
+
+func getStatusForRPCBehaviorMetadata(headerValue string) *status.Status {
+	if !strings.HasPrefix(headerValue, xdsinterop.ErrorCodePrefix) {
+		return nil
+	} else if errCode, err := strconv.Atoi(headerValue[len(xdsinterop.ErrorCodePrefix):]); err != nil {
+		return status.New(codes.InvalidArgument, "Invalid format for rpc-behavior header: "+headerValue)
+	} else {
+		return status.New(codes.Code(errCode), "Rpc failed as per the rpc-behavior header value: "+headerValue)
```

Comment:
Same here, please you formatting directives. Also, would prefer keeping the happy case in the if (i.e. switch if conditional to err == nil).

Reply:
You can find more info here: https://google.github.io/styleguide/go/decisions#indent-error-flow